lib/rules/disallow-parentheses-around-arrow-param.js

```diff
@@ -58,21 +58,22 @@ module.exports.prototype = {
             if (node.params.length !== 1) {
                 return;
             }
+            var firstParam = node.params[0];
 
             // Old Esprima
             var hasDefaultParameter = node.defaults && node.defaults.length === 1;
             // ESTree
-            var hasDefaultParameterESTree = node.params[0].type === 'AssignmentPattern';
-            var hasDestructuring = node.params[0].type === 'ObjectPattern';
+            var hasDefaultParameterESTree = firstParam.type === 'AssignmentPattern';
+            var hasDestructuring = firstParam.type === 'ObjectPattern' || firstParam.type === 'ArrayPattern';
+            var hasRestElement = firstParam.type === 'RestElement';
 
```

Comment:
Maybe it would be easier to just check if the first param is not an identifier (and return) instead of all the checks?